apps/src/code-studio/pd/application/teacher1819/Section4SummerWorkshop.jsx

```diff
@@ -0,0 +1,356 @@
+
+    this.loadPartnerWorkshops();
+    if (this.isUnableToAttendAssignedWorkshop()) {
+      this.loadAlternateWorkshops();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.loadPartnerRequest) {
+      this.loadPartnerRequest.abort();
+    }
+    if (this.loadAlternateWorkshopsRequest) {
+      this.loadAlternateWorkshopsRequest.abort();
+    }
+  }
+
+  loadPartnerWorkshops() {
+    const locationParams = this.getWorkshopParams();
+    if (this.props.data.school === '-1') {
+      locationParams.zip_code = this.props.data.schoolZipCode;
+      locationParams.state = this.props.data.schoolState;
+    } else {
+      locationParams.school = this.props.data.school;
+    }
+
+    const url = `/api/v1/pd/regional_partner_workshops/find?${$.param(locationParams)}`;
+    this.loadPartnerRequest = $.ajax({
+      method: 'GET',
+      url: url,
+      dataType: 'json'
+    }).done(data => {
+      this.loadPartnerRequest = null;
+
+      // Persist in the form data
+      this.handleChange({
+        regionalPartnerId: data.id,
+        regionalPartnerGroup: data.group,
+        regionalPartnerWorkshopCount: data.workshops ? data.workshops.length : 0
+      });
+
+      // Update state with all the partner workshop data to display
+      this.setState({
+        loadingPartner: false,
+        partnerWorkshops: data.workshops
+      });
+    });
+  }
+
+  loadAlternateWorkshops() {
+    const url = `/api/v1/pd/regional_partner_workshops?${$.param(this.getWorkshopParams())}`;
+    this.loadAlternateWorkshopsRequest = $.ajax({
+      method: 'GET',
+      url: url,
+      dataType: 'json'
+    }).done(data => {
+      this.loadAlternateWorkshopsRequest = null;
+      const alternateWorkshops = data.reduce((workshops, partner) => (
+        partner.id !== this.props.data.regionalPartnerId
+          ? workshops.concat(partner.workshops)
+          : workshops
+      ), []);
+
+      this.setState({
+        loadingAlternateWorkshops: false,
+        alternateWorkshops
+      });
+    });
+  }
+
+  getWorkshopParams() {
+    const course = {
+      [PROGRAM_CSD] : 'CS Discoveries',
+      [PROGRAM_CSP] : 'CS Principles'
+    }[this.props.data.program];
+
+    return {
+      course,
+      subject: '5-day Summer'
+    };
+  }
+
+  isUnableToAttendAssignedWorkshop = (data = this.props.data) => data.regionalPartnerId && (
+      data.ableToAttendSingle === UNABLE_TO_ATTEND ||
+      (
+        data.ableToAttendMultiple &&
+        data.ableToAttendMultiple.length === 1 &&
+        data.ableToAttendMultiple[0] === NO_EXPLAIN
+      )
+    );
+
+  /**
+   * @override
+   */
+  handleChange(newState) {
+    const data = {...this.props.data, ...newState};
+    if (this.isUnableToAttendAssignedWorkshop(data) && !this.state.alternateWorkshops && !this.loadAlternateWorkshopsRequest) {
+      this.loadAlternateWorkshops();
+    }
+
+    super.handleChange(newState);
+  }
+
+  handleUnderstandFeeChange = event => {
```

Comment:
Optional (and maybe overkill) but you can combine these two functions into one. If you give the controls respective ids of `understandFee` and `considerForFunding` you can just have one function like
```
handleConsiderForFundingChange = event => {
     this.handleChange({
      [event.target.id]: event.target.checked
     });
   };
```

Reply:
Actually neither of these are needed. They are leftover from before I added the [SingleCheckbox](https://github.com/code-dot-org/code-dot-org/blob/staging-next/apps/src/code-studio/pd/form_components/SingleCheckbox.jsx) form component, which internally handles the update, and is wired up with `this.handleChange` in the base FormComponent's [buildSingleCheckbox](https://github.com/code-dot-org/code-dot-org/blob/staging-next/apps/src/code-studio/pd/form_components/FormComponent.jsx#L300) method.

Will remove.